cli/executor.h

```diff
@@ -70,7 +70,8 @@ class Executor {
 
 private:
     std::mutex mErrorListSync;
-    std::list<std::string> mErrorList;
+    // TODO: store hashes instead of full messages
+    std::unordered_set<std::string> mErrorList;
```

Comment:
Spontanously it feels to me that `std::set` would be faster. Every time we want to check if a message already exists the unordered_set must loop through all items right?

But well I assume you have measured the performance? Did you try a use case when there was many error messages?

Reply:
> But well I assume you have measured the performance? Did you try a use case when there was many error messages?

Yes. The case mentioned above generates a lot of messages that's why I chose it. I tested `std::list`, `std::vector`, `std::set` and `std::unordered_set` and the latter was the fastest.